internal/services/report_processor.go

```diff
@@ -75,122 +75,138 @@ func ProcessReport(msg *kafka.Message) {
 
-		keys := make([]string, 0, len(k8s_object_groups))
-		for key := range k8s_object_groups {
-			keys = append(keys, key)
-		}
+		for _, v := range k8s_object_groups {
 
-		sort.SliceStable(keys, func(i, j int) bool {
-			time_i, _ := utils.ConvertStringToTime(k8s_object_groups[keys[i]].Col("interval_end").Val(0).(string))
-			time_j, _ := utils.ConvertStringToTime(k8s_object_groups[keys[j]].Col("interval_end").Val(0).(string))
-			return time_i.Before(time_j)
-		})
-
-		// looping over each group in chronological order based on interval_end time.
-		for _, group_name := range keys {
-			k8s_object := k8s_object_groups[group_name].Maps()
-			namespace := k8s_object[0]["namespace"].(string)
-			k8s_object_type := k8s_object[0]["k8s_object_type"].(string)
-			k8s_object_name := k8s_object[0]["k8s_object_name"].(string)
-			interval_start, err := utils.ConvertStringToTime(k8s_object[0]["interval_start"].(string))
-			if err != nil {
-				log.Errorf("unable to convert string to time: %s", err)
-				continue
-			}
-			interval_end, err := utils.ConvertStringToTime(k8s_object[0]["interval_end"].(string))
+			all_interval_end_time := v.Col("interval_end").Records()
+			maxEndTime, err := utils.MaxIntervalEndTime(all_interval_end_time)
 			if err != nil {
 				log.Errorf("unable to convert string to time: %s", err)
 				continue
 			}
+			k8s_object_group_by_interval_end := v.GroupBy("interval_end").GetGroups()
 
-			experiment_name := utils.GenerateExperimentName(
-				kafkaMsg.Metadata.Org_id,
-				kafkaMsg.Metadata.Source_id,
-				kafkaMsg.Metadata.Cluster_uuid,
-				namespace,
-				k8s_object_type,
-				k8s_object_name,
-			)
-
-			if workload_metrics, err := model.GetWorkloadMetricsForTimestamp(experiment_name, interval_end); err != nil {
-				log.Errorf("Error while checking for workload_metrics record: %s", err)
-				continue
-			} else if !reflect.ValueOf(workload_metrics).IsZero() {
-				log.Debugf("worload_metrics table already has data for interval_end time: %v.", interval_end)
-				continue
-			}
-
-			container_names, err := kruize.Create_kruize_experiments(experiment_name, k8s_object)
-			if err != nil {
-				log.Error(err)
-				continue
+			keys := make([]string, 0, len(k8s_object_group_by_interval_end))
+			for key := range k8s_object_group_by_interval_end {
+				keys = append(keys, key)
 			}
 
-			// Create workload entry into the table.
-			workload := model.Workload{
-				ClusterID:       cluster.ID,
-				ExperimentName:  experiment_name,
-				Namespace:       namespace,
-				WorkloadType:    workload.WorkloadType(k8s_object_type),
-				WorkloadName:    k8s_object_name,
-				Containers:      container_names,
-				MetricsUploadAt: time.Now(),
-			}
-			if err := workload.CreateWorkload(); err != nil {
-				log.Errorf("unable to get or add record to workloads table: %v. Error: %v", workload, err)
-				continue
-			}
+			sort.SliceStable(keys, func(i, j int) bool {
```

Comment:
```suggestion
			sort.SlicesTable(keys, func(i, j int) bool {
```
@patilsuraj767 Could you add a comment here, especially regarding the purpose of `Before`?

Reply:
`SliceStable` is a function in `sort` package we cannot change the function name.  Using `sort.SliceStable` we are basically sorting the group of deployment into chronological order based on `interval_end` time